System.cpp

```diff
@@ -271,3 +298,45 @@ void System::echo(std::string content, std::string fname){
         }
     }
 }
+
+void System::mv(std::string name, std::string dest) {
+	if (alreadyExists(name)) {
+		Node* toMove = stringToNode(name);
+		if (alreadyExists(dest)) {
+			Node* destination = stringToNode(dest);
+			if (dynamic_cast<Dir*>(destination) != nullptr) {
+                //TODO: absolut path esetén nme törli ki!!!
+                Dir* current = currentFolder;
+                std::string working;
+                
+                try {
+                    working = goToFolder(name);
+                } catch (NoDirectoryExc e) {
+                    std::cerr << e.getWhat() << std::endl;
+                }
+                
+                std::vector<Node*>tmp  = currentFolder->getChildren();
+                for (unsigned i = 0; i < tmp.size(); ++i){
+                    if (tmp[i]->getName() == working) {
+                        tmp.erase(tmp.begin() + i);
+                    }
+                }
+                currentFolder->setChildren(tmp);
+                currentFolder = current;
```

Comment:
Kisse erthetetlen a kod (mintha mar mashol lattam volna amugy)

de van egy current, amit atallitunk egy currentfolderre, majd a currentfolder  gyerekeit lemasoljuk, kitoroljuk a working-et, es a madradekot visszamasoljuk (why nem eleve az eredetin toroltunk?!) aztan utana a currentfoldert atallitjuk a current-re, ami meg mindig acurrentcolder, mert azota hozza sem nyultunk. 

wut?

Reply:
ohhh szerintem az a baj hogy ezt a branchet a mv commandos brancbol származtattam. ezért belevette ebbe azokat a commitokat is. Erre valami megoldás? Vagy töröljük várjuk meg mig a mv-t be mergeled és utána csináljuk?

Reply:
Hat... lekoteleznetek vele, hogyha nem kellene ketszer ugyanazt megcsinalni :-) De amguy az itt megirtak ettol fuggetlenul valid eszrevetelek, szoval ha azon a branchen sincs rendben, erdemes ott javitani. Ezt meg akar rebase-elni ra majd.